<!DOCTYPE html>
<html>
<head>
  <title>Microbial Informatics</title>
  <meta charset="utf-8">
  <meta name="description" content="Microbial Informatics">
  <meta name="author" content="Patrick D. Schloss, PhD (microbialinformatics.github.io)">
  <meta name="generator" content="slidify" />
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta http-equiv="X-UA-Compatible" content="chrome=1">
  <link rel="stylesheet" href="http://slidifylibraries2.googlecode.com/git/inst/libraries/frameworks/io2012/css/default.css" media="all" >
  <link rel="stylesheet" href="http://slidifylibraries2.googlecode.com/git/inst/libraries/frameworks/io2012/css/phone.css" 
    media="only screen and (max-device-width: 480px)" >
  <link rel="stylesheet" href="http://slidifylibraries2.googlecode.com/git/inst/libraries/frameworks/io2012/css/slidify.css" >
  <link rel="stylesheet" href="http://slidifylibraries2.googlecode.com/git/inst/libraries/highlighters/highlight.js/css/tomorrow.css" />
  <base target="_blank"> <!-- This amazingness opens all links in a new tab. -->  <link rel=stylesheet href="./assets/css/my.css"></link>
<link rel=stylesheet href="./assets/css/ribbons.css"></link>
<link rel=stylesheet href="./assets/css/slidify.css"></link>

  
  <!-- Grab CDN jQuery, fall back to local if offline -->
  <script src="http://ajax.aspnetcdn.com/ajax/jQuery/jquery-1.7.min.js"></script>
  <script>window.jQuery || document.write('<script src="http://slidifylibraries2.googlecode.com/git/inst/libraries/widgets/quiz/js/jquery.js"><\/script>')</script> 
  <script data-main="http://slidifylibraries2.googlecode.com/git/inst/libraries/frameworks/io2012/js/slides" 
    src="http://slidifylibraries2.googlecode.com/git/inst/libraries/frameworks/io2012/js/require-1.0.8.min.js">
  </script>
  
  

</head>
<body style="opacity: 0">
  <slides class="layout-widescreen">
    
    <!-- LOGO SLIDE -->
        <slide class="title-slide segue nobackground">
  <hgroup class="auto-fadein">
    <h1>Microbial Informatics</h1>
    <h2>Lecture 05</h2>
    <p>Patrick D. Schloss, PhD (microbialinformatics.github.io)<br/>Department of Microbiology &amp; Immunology</p>
  </hgroup>
  <article></article>  
</slide>
    

    <!-- SLIDES -->
    <slide class="" id="slide-1" style="background:;">
  <hgroup>
    <h2>Announcements</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Start thinking about your project for the first half of the semester

<ul>
<li>Emphasis on data analysis</li>
<li>Due 10/24/2104 (friday)</li>
<li>Feel free to come to office hours to discuss project ideas</li>
<li>I have some ideas for microbial ecology analysis projects</li>
</ul></li>
<li>When you upload your assignments, upload the README.Rmd and README.md files generated by RStudio/knitr</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-2" style="background:;">
  <hgroup>
    <h2>Review</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Everything in R is some form of a vector</li>
<li>Vectors</li>
<li>List</li>
<li>Matrix</li>
<li>Table</li>
<li>Data table</li>
<li>Factors</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-3" style="background:;">
  <hgroup>
    <h2>Learning objectives</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Appreciate the diversity of different data structures</li>
<li>Understand how to calculate descriptive variables</li>
<li>Begin to develop data visualization skills in R</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-4" style="background:;">
  <hgroup>
    <h2>Factors</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Defining categorical variables</li>
<li>In a genome we might think of the forward/reverse orientation, reading frame, dna/protein sequence designation, or annotation category as categorical variables.<br></li>
<li>Create factors</li>
</ul>

<pre><code class="r">metadata &lt;- read.table(file=&quot;wild.metadata.txt&quot;, header=T)
rownames(metadata) &lt;- metadata$Group
metadata &lt;- metadata[,-1]

factor(metadata$ET)
metadata$ET &lt;-factor(metadata$ET)
summary(metadata)
levels(metadata$ET)
</code></pre>

<ul>
<li>What other variables here would be a factor?</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-5" style="background:;">
  <hgroup>
    <h2>Lists</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Similar to data fames, but not necessarily read across rows and not all variables have the same length</li>
<li>Could hold a genome&#39;s data within a list:

<ul>
<li>name: Character with organism name</li>
<li>genome.size: Number with number of bases</li>
<li>start.pos: Vector of start positions for each gene</li>
<li>end.pos: Vector of end positions for each gene</li>
<li>gene.name: Name of each gene</li>
<li>hydrolases: Names of genes that are hydrolases</li>
</ul></li>
<li>Allow one to create complex data structures</li>
<li>We&#39;ll use these only in passing</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-6" style="background:;">
  <hgroup>
    <h2>An example of where we&#39;ll use lists...</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Let&#39;s get the mean weight for each sex of mouse</li>
</ul>

<pre><code class="r">aggregate(metadata$Weight, by=metadata$Sex, mean)
aggregate(metadata$Weight, by=list(metadata$Sex), mean)
sex.weight &lt;- aggregate(metadata$Weight, by=list(metadata$Sex), mean)
sex.weight$x
</code></pre>

<ul>
<li>Let&#39;s get the mean weight for each sex and species of mouse</li>
</ul>

<pre><code class="r">aggregate(metadata$Weight, by=list(metadata$Sex, metadata$SP), mean)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-7" style="background:;">
  <hgroup>
    <h2>Descriptive data analysis</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>  With large datasets it&#39;s easy to get lost and not know what the data look like</li>
<li>  Need to devlop strategies to get a handle on the data</li>
<li>  Three approaches:

<ul>
<li>Visualize data tables</li>
<li>Summary statistics</li>
<li>Visualize data</li>
</ul></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-8" style="background:;">
  <hgroup>
    <h2>What types of variables do we have in the table?</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>How do we get the column/variable names?</li>
<li>How do we know the data type for each column?</li>
<li>Which columns in <code>metadata</code> should be<br>

<ul>
<li>  Continuous?</li>
<li>  Categorical?</li>
</ul></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-9" style="background:;">
  <hgroup>
    <h2>Visualizing raw data</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>  Homework asked you to print out the first 5 lines of a data table</li>
<li>There ares everal ways to do this...</li>
</ul>

<pre><code class="r">datatable[c(1,2,3,4,5),]    # explicit method   
datatable[1:5,]             # bracket method
head(datatable, n=5)                # first N lines
tail(datatable, n=5)                # last N lines
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-10" style="background:;">
  <hgroup>
    <h2>Summary statistics</h2>
  </hgroup>
  <article data-timings="">
    <p>What types of data structures do these output?  </p>

<pre><code class="r">summary(metadata)           # how do we get a summary of the overall metadata table?
summary(metadata$Weight)    # how do we get a summary of the weights?
median(metadata$Weight)     # median - what is a median?
w.summary &lt;- summary(metadata$Weight)
w.summary[&quot;Median&quot;]

sd(metadata$Weight)         # standard deviation
var(metadata$Weight)        # variance
min(metadata$Weight)        # minimum weight
max(metadata$Weight)        # maximum weight
quantile(metadata$Weight)   # weight distribution by quantiles                          
quantile(metadata$Weight, probs=seq(0,1,0.1))   # weight distribution by deciles
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-11" style="background:;">
  <hgroup>
    <h2>Let&#39;s cross reference columns</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">aggregate(metadata$Weight, by=list(metadata$Sex), mean)
aggregate(Weight~Sex, data=metadata, mean)  #Notice the difference in output

aggregate(metadata$Weight, by=list(metadata$Sex, metadata$SP), mean)
aggregate(Weight~Sex+SP, data=metadata, mean)

aggregate(Weight~Sex+SP, data=metadata, summary)

w.sex_sp &lt;- aggregate(Weight~Sex+SP, data=metadata, mean)   # what type of variable is w.sex_sp?
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-12" style="background:;">
  <hgroup>
    <h2>Data visualization</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>This is a huge area of explorataion</li>
<li>R is tremendously powerful for generating plots and data visualizaiton tools</li>
<li>Can generally tell someone used MS Excel by how bad the plots look</li>
<li>Can certainly generate crap in R, but upside is greater</li>
<li>Numerous packages available, but we will focus on base package until the end of the semester:

<ul>
<li>Lattice</li>
<li>ggplot2</li>
<li>rgl</li>
</ul></li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-13" style="background:;">
  <hgroup>
    <h2>Histograms</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Good for summarizing continuous data where you want to break it into discrete classes</li>
<li>What do these two sets of commands do?</li>
</ul>

<pre><code class="r">par(mfrow = c(2, 1))  # make two plotting windows
hist(metadata$Weight[metadata$Sex == &quot;F&quot;])
hist(metadata$Weight[metadata$Sex == &quot;M&quot;])
par(mfrow = c(1, 1))  # return to one plotting window

par(mfrow = c(2, 1))
hist(metadata$Weight[metadata$Sex == &quot;F&quot;], breaks = 10, ylim = c(0, 20), xlim = c(0, 
    30))
hist(metadata$Weight[metadata$Sex == &quot;M&quot;], breaks = 10, ylim = c(0, 20), xlim = c(0, 
    30), add = T)
par(mfrow = c(1, 1))
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-14" style="background:;">
  <hgroup>
    <h2>Merging plots with <code>add=T</code></h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">hist(metadata$Weight[metadata$Sex == &quot;F&quot;], breaks = 10, ylim = c(0, 20), xlim = c(0, 
    30), col = &quot;pink&quot;)
hist(metadata$Weight[metadata$Sex == &quot;M&quot;], breaks = 10, col = &quot;blue&quot;, add = T)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-15" style="background:;">
  <hgroup>
    <h2>Universal options</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Axis labels</li>
</ul>

<pre><code class="r">hist(metadata$Weight, xlab=&quot;Weights of Peromyscus spp.&quot;)
</code></pre>

<ul>
<li>Plot title</li>
</ul>

<pre><code class="r">hist(metadata$Weight, main=&quot;Distribution of Peromyscus spp. weights&quot;)
</code></pre>

<ul>
<li>Putting it together</li>
</ul>

<pre><code class="r">hist(metadata$Weight, main = &quot;Distribution of Peromyscus spp. weights&quot;, xlab = &quot;Weights of Peromyscus spp.&quot;)
box()
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-16" style="background:;">
  <hgroup>
    <h2>What is the output from <code>hist</code>?</h2>
  </hgroup>
  <article data-timings="">
    <pre><code class="r">m.hist &lt;- hist(metadata$Weight[metadata$Sex == &quot;F&quot;], breaks = 10, ylim = c(0, 
    20), xlim = c(0, 30), col = &quot;pink&quot;)
f.hist &lt;- hist(metadata$Weight[metadata$Sex == &quot;M&quot;], breaks = 10, col = &quot;blue&quot;, 
    add = T)
</code></pre>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="" id="slide-17" style="background:;">
  <hgroup>
    <h2>For Thursday</h2>
  </hgroup>
  <article data-timings="">
    <ul>
<li>Start working on new assignment that will be posted this weekend</li>
<li>Assignment due Friday</li>
<li>Read <strong><em>Introduction to Statistics with R</em></strong> (Chapter 3 and 4)</li>
</ul>

  </article>
  <!-- Presenter Notes -->
</slide>

<slide class="segue dark" id="slide-18" style="background:;">
  <hgroup>
    <h2>Questions?</h2>
  </hgroup>
  <article data-timings="">
    
  </article>
  <!-- Presenter Notes -->
</slide>

    <slide class="backdrop"></slide>
  </slides>
  <div class="pagination pagination-small" id='io2012-ptoc' style="display:none;">
    <ul>
      <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=1 title='Announcements'>
         1
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=2 title='Review'>
         2
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=3 title='Learning objectives'>
         3
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=4 title='Factors'>
         4
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=5 title='Lists'>
         5
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=6 title='An example of where we&#39;ll use lists...'>
         6
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=7 title='Descriptive data analysis'>
         7
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=8 title='What types of variables do we have in the table?'>
         8
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=9 title='Visualizing raw data'>
         9
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=10 title='Summary statistics'>
         10
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=11 title='Let&#39;s cross reference columns'>
         11
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=12 title='Data visualization'>
         12
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=13 title='Histograms'>
         13
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=14 title='Merging plots with <code>add=T</code>'>
         14
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=15 title='Universal options'>
         15
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=16 title='What is the output from <code>hist</code>?'>
         16
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=17 title='For Thursday'>
         17
      </a>
    </li>
    <li>
      <a href="#" target="_self" rel='tooltip' 
        data-slide=18 title='Questions?'>
         18
      </a>
    </li>
  </ul>
  </div>  <!--[if IE]>
    <script 
      src="http://ajax.googleapis.com/ajax/libs/chrome-frame/1/CFInstall.min.js">  
    </script>
    <script>CFInstall.check({mode: 'overlay'});</script>
  <![endif]-->
</body>
  <!-- Load Javascripts for Widgets -->
  
  <!-- LOAD HIGHLIGHTER JS FILES -->
  <script src="http://slidifylibraries2.googlecode.com/git/inst/libraries/highlighters/highlight.js/highlight.pack.js"></script>
  <script>hljs.initHighlightingOnLoad();</script>
  <!-- DONE LOADING HIGHLIGHTER JS FILES -->
   
  </html>